src/surface/volume.ts

```diff
@@ -293,7 +293,7 @@ class Volume {
 
     const box = this._getBox(center, size)
     const sd = this.volsurf.getSurface(
-      isolevel, smooth, box!, this.matrix.elements, contour, wrap
+      isolevel, smooth, box!, new Float32Array(this.matrix.elements), contour, wrap
```

Comment:
I guess you are passing `matrix.elements` to the Float32Array constructor here to avoid TS complaints. There is a perf penalty in that statement. As it does not change the way the code runs, I'd suggest forcing the typing here with a `this.matrix.elements as unknown as Float32Array`
Same goes for l:382

Reply:
Good point about performance. But I don't think your cast idea will work, because `this.matrix.elements` is actually an array of doubles ("number[]"), so you can't just cast it to `Float32Array`, right? It would look like garbage. And a `matrix` is pretty small, so the copy/convert overhead shouldn't be *that* bad.

Reply:
Well, I wrote this whole comment about how that couldn't possibly work because `matrix` is a `number[]` so it's doubles, and `Float32Array` is floats, so the cast should come out as garbage (can you tell I'm an old-school C/C++ guy?) but I did a simple standalone test and it does work. By "work" I mean the generated JS ignores all the types, so it ends up passing an array of doubles into `getSurface` which is expecting floats, but as long as that's OK all the way down, I guess that's OK...

Reply:
Yes, that was my guess too. The code was written first with these careful assumptions. As long as we don't meddle with the underlying js logic, it should continue to work Ok.